mock-core-configs/etc/mock/templates/epel-10.tpl

```diff
@@ -2,10 +2,35 @@ config_opts['chroot_setup_cmd'] += " epel-rpm-macros"
 
 config_opts['dnf.conf'] += """
 
+# The metalinks below differ from the ones used in epel-release.  The
+# epel-release metalinks rely on a dnf variable expansion feature that is only
+# available since dnf 4.18.0.  In some configurations mock will be using the
+# host dnf to bootstrap the chroot, which could be older and lack those
+# features, leading to problems.  The alternative is to use this simplified
+# metalink here in this template for CentOS Stream, and a different metalink in
+# another template for RHEL.
+
+[epel]
+name=Extra Packages for Enterprise Linux $releasever - $basearch
+metalink=https://mirrors.fedoraproject.org/metalink?repo=epel-{{ releasever_major }}&arch=$basearch
+gpgkey=file:///usr/share/distribution-gpg-keys/epel/RPM-GPG-KEY-EPEL-{{ releasever_major }}
+gpgcheck=1
+countme=1
+
+[epel-testing]
+name=Extra Packages for Enterprise Linux $releasever - Testing - $basearch
+metalink=https://mirrors.fedoraproject.org/metalink?repo=testing-epel{{ releasever_major }}&arch=$basearch
+gpgkey=file:///usr/share/distribution-gpg-keys/epel/RPM-GPG-KEY-EPEL-{{ releasever_major }}
+gpgcheck=1
+countme=1
+enabled=0
+
+{% if koji_primary_repo == "epel" %}
 [local]
 name=Extra Packages for Enterprise Linux $releasever - Koji Local - BUILDROOT ONLY!
-baseurl=https://kojipkgs.fedoraproject.org/repos/epel10.0-build/latest/$basearch/
+baseurl=https://kojipkgs.fedoraproject.org/repos/epel{{ releasever_major }}/centos/$basearch/
```

Comment:
Is this a floating symlink to some other repository eventually?  Can you perhaps document the trick here, or link a documentation that documents it?

Reply:
It's a symlink in the `/mnt/fedora_koji` network share in the Fedora infrastructure, similar to `rawhide/latest`.

```
[carlwgeorge@compose-rawhide01 repos][PROD-IAD2]$ pwd
/mnt/fedora_koji/koji/repos
[carlwgeorge@compose-rawhide01 repos][PROD-IAD2]$ ll rawhide/latest
lrwxrwxrwx. 1 root root 19 Aug 13 23:52 rawhide/latest -> ../f42-build/latest
[carlwgeorge@compose-rawhide01 repos][PROD-IAD2]$ ll epel10/centos
lrwxrwxrwx. 1 root root 24 Oct  9 01:54 epel10/centos -> ../epel10.0-build/latest
```

My thinking on the symlink naming is that it would be best to use the names centos and rhel.  In the other issues we've thrown around name pairs like development+stable and latest+non-latest, but none of those sit quite right with me.  It's the latest EPEL for CentOS and the latest EPEL for RHEL, so I think the symlinks should just match that.  In the future, we'll have `epel10/centos -> ../epel10.1-build/latest` and `epel10/rhel -> ../epel10.0-build/latest`, and they will move forward over time.

I don't have this written down in formal documentation yet, but my team (CPE EPEL) is tracking our EPEL 10 work in some hackmd docs, which will serve as the basis for a future SOP in the releng and/or EPEL docs.  I've got steps in there for adjusting these symlinks over time.

- [EPEL 10.0 work tracker](https://hackmd.io/@carlwgeorge/ry_xGk35C)
- [EPEL 10.1 work tracker](https://hackmd.io/@carlwgeorge/Sk3j-12qA)

I can add a comment above this line with a basic summary that the link is a symlink that will move forward over time.

Reply:
Thank you for the update! Makes sense.